### ISSUE TITLE
Add ability to pause all animated content within the editor

### ISSUE THREAD

**Issue by Zireael07** [godotengine/godot-proposals#942]
**Describe the project you are working on:**
3d open world game

**Describe the problem or limitation you are having in your project:**
I have many particle emitters in the world. They have a noticeable effect on the in-editor FPS.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Pausing particle simulation in editor lets me work without FPS drops.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
A button/switch in project/editor settings that pauses ALL animated content in editor.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
While this is possible by setting emitting to false on particles, it requires attaching a tool script to EVERY particle emitter. The bigger the project, the less feasible this gets.

**Is there a reason why this should be core and not an add-on in the asset library?:**
How would an asset know what and when to pause? IIRC it needs core changes, plus assets can't add to project settings AFAIK.

Original issue: https://github.com/godotengine/godot/issues/16397

ETA: It seems that the animated sprites and tiles suffer from the same problem, and especially for tiles, the workaround would be very involved.

**Comment by itsjavi** [comment]
I would extend this to "any kind of animated content".

Related: 

https://github.com/godotengine/godot/issues/39758
https://github.com/godotengine/godot/pull/31070

**Comment by chucklepie** [comment]
Can this be updated to include animated tiles and animations using tres as the same thing happens. I raised ticket #1425 for this, but it was closed for 'duplicate' of this. Thanks.

**Comment by Zireael07** [comment]
No problem, someone above you already mentioned this should refer to all animated content - I will update OP to make it clearer.

**Comment by akien-mga** [comment]
Implemented by https://github.com/godotengine/godot/pull/53463.

**Comment by QbieShay** [comment]
Was this implemented for 4.x?

**Comment by Calinou** [comment]
> Was this implemented for 4.x?

No, https://github.com/godotengine/godot/pull/53463 hasn't been forward-ported to `master` yet.

**Comment by QbieShay** [comment]
reopening for 4.x

**Comment by luislodosm** [comment]
I agree. Currently, a simple particle system can have a very high demand on the GPU. I think the default behaviour could be: 

- Particle nodes are **paused** in the editor by default.
- If the particle node is selected, then it starts simulating.
- A "**particle group**" node could be used to control the state of children particle nodes.
- If "**simulate all particle nodes**" button is enabled, all particles of the scene simulated.

**Comment by QbieShay** [comment]
@luislodosm i really need multiple particle system to play at the same time in editor in order to be able to create VFX. whatever the design of this feature, it shouldn't come at the cost of usability of everyone's VFX. You could have a toolscript that does that, if you have a need for this kind of feature.

**Comment by Zireael07** [comment]
How does pausing animated content in editor hurt people's VFX?

Note this was already implemented in 3.5, too

**Comment by DiMartino49** [comment]
Has this been implemented in godot 4.x?

**Comment by Calinou** [comment]
> Has this been implemented in godot 4.x?

No, not yet.

**Comment by pjhenning** [comment]
Hey just chiming in to say that I have a somewhat underpowered computer, would love to give Godot a shot but its pretty unusable for me without this. Even the bloated Unreal Engine 5 is more practical for me to work with currently because it provides the option to pause rendering entirely when not adjusting the camera view.
